### PR TITLE
Mitigation Round Code and Example

### PR DESCRIPTION
**Pull request by jkcarney**
This is a minimal example for our mitigation round code. It supports two entry points, `--mitigate` and `--test` which respectively runs the mitigation algorithm and saves, and then tests the mitigation algorithm by running inference and saving the logits. It supports a metaparameters file that is baked into the image and acts as defaults. The Dockerfile builds and runs with some caveats (see below). 

This is not 100% feature complete;
- Ideally the dataset passed in has no label available. We're creating a type of dataset that makes it easier to bookkeep what data is clean/poisoned as well as the label while also making sure that label isn't available to the container.
- This Dockerfile assumes that two APL internal dependencies are cloned and available locally; a BackdoorBox wrapper, and our latest code for the mitigation round. Currently, the code available in the private round generation repo is out of date and does not contain this baseline as well as other updates to the code
  - As an extension of this, we would like to be able to develop the mitigation round code directly in the private round generation repository, so we would need to be granted access to make branches.

### CODE REVIEW

File: .gitmodules
Comment (mmajurski) [inline .gitmodules:1]: remove the submodule trojai-round-generation-private.
